#!/bin/sh /usr/share/dpatch/dpatch-run

## 18_vdr-1.3.33-recordingsmenu.dpatch by Klaus Schmidinger <kls@cadsoft.de>
## from vdr@linuxtv.org mailinglist (Sat, 01 Oct 2005 12:16:44 +0200)
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes an error after deleting a recording from a subfolder.

@DPATCH@
--- vdr-1.3.33/menu.c	2005/09/25 13:37:21	1.362
+++ vdr-1.3.33/menu.c	2005/10/01 10:12:32
@@ -1513,6 +1513,7 @@
      SetCurrent(First());
   else if (OpenSubMenus && cReplayControl::LastReplayed() && Open(true))
      return;
+  Display();
   SetHelpKeys();
 }
 
@@ -1556,7 +1557,7 @@
   if (Refresh) {
      cMenuRecordingItem *ri = (cMenuRecordingItem *)Get(Current());
      if (ri) {
-        cRecording *Recording = GetRecording(ri);
+        cRecording *Recording = Recordings.GetByName(ri->FileName());
         if (Recording)
            CurrentRecording = Recording->FileName();
         }
@@ -1583,7 +1584,8 @@
          }
       }
   free(LastItemText);
-  Display();
+  if (Refresh)
+     Display();
 }
 
 cRecording *cMenuRecordings::GetRecording(cMenuRecordingItem *Item)

